Sources/Commands/SwiftTool.swift

```diff
@@ -561,6 +594,23 @@ public class SwiftTool<Options: ToolOptions> {
     func runLLBuild(manifest: AbsolutePath, llbuildTarget: String) throws {
         assert(localFileSystem.isFile(manifest), "llbuild manifest not present: \(manifest.asString)")
         
+        if options.shouldEnableLLBuildLibrary {
+            runLLBuildAsLibrary(manifest: manifest, llbuildTarget: llbuildTarget)
+        } else {
+            runLLBuildAsExecutable(manifest: manifest, llbuildTarget: llbuildTarget)
+        }
+    }
+
+    func runLLBuildAsLibrary(manifest: AbsolutePath, llbuildTarget: String) throws {
+        let buildSystem = BuildSystem(buildFile: manifest.asString, databaseFile: "build.db", delegate: self)
```

Comment:
I would recommend factoring out a separate class responsible for handling this and being the delegate. It should be possible to unit test that thing in isolation from the rest of the SwiftTool logic.

Reply:
Were you thinking of a `LLBuildTool` protocol with two implementations `LLBuildExecutableTool` and `LLBuildLibraryTool`?